Source/Core/VideoBackends/Vulkan/Renderer.cpp

```diff
@@ -650,71 +651,91 @@ void Renderer::BlitScreen(VkRenderPass render_pass, const TargetRectangle& dst_r
 
 void Renderer::CheckForSurfaceChange()
 {
-  if (!m_surface_needs_change.IsSet())
+  if (!m_surface_needs_change.TestAndClear())
     return;
 
-  // Wait for the GPU to catch up since we're going to destroy the swap chain.
+  void* new_surface_handle;
+  {
+    std::lock_guard<std::mutex> guard(m_surface_change_mutex);
+    new_surface_handle = m_new_surface_handle;
+    m_new_surface_handle = nullptr;
+    m_surface_handle = m_new_surface_handle;
```

Comment:
is m_surface_handle = nullptr on purpose?

Reply:
Nope, that's definitely incorrect. Thanks for catching it.